docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,297 @@
+2024-06-28T02:17:24.363Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:17:25.370Z [pebble] Service "frontend-server" starting: python3 -m http.server 8080
+2024-06-28T02:17:26.385Z [pebble] Started default services with change 1.
+```
+
+We can see they are all started, which can be verified by:
+
+```{terminal}
+   :input: pebble services
+Service          Startup  Current  Since
+backend-server   enabled  active   today at 10:17 CST
+database         enabled  active   today at 10:17 CST
+frontend-server  enabled  active   today at 10:17 CST
+```
+
+## Problems
+
+Apparently, the above layer configuration works but is not exactly ideal, because:
+
+- In this example, the three components kind of rely on each other, without anyone, our whole web app won't function properly.
+- If the database isn't started (or can't be started for some reason), it doesn't make sense to start the backend that depends on it. Even if the backend starts, it fails to communicate with the database, so the whole web app won't function properly.
+- The same is true for the frontend: if the backend isn't started (fails to start, can't be started, etc.), it makes no sense to start the frontend alone.
+
+For example, when the DB fails to start, other services are still up and running:
+
+```{terminal}
+   :input: pebble run
+2024-06-28T02:20:03.337Z [pebble] Started daemon.
+[backend-server database frontend-server]
+2024-06-28T02:20:03.343Z [pebble] POST /v1/services 2.763792ms 202
+2024-06-28T02:20:03.346Z [pebble] Service "backend-server" starting: python3 -m http.server 8081
+2024-06-28T02:20:03.346Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:20:03.347Z [pebble] Service "frontend-server" starting: python3 -m http.server 8080
+2024-06-28T02:20:03.396Z [pebble] Change 1 task (Start service "database") failed: cannot start service: exited quickly with code 1
+2024-06-28T02:20:04.353Z [pebble] Started default services with change 1.
+```
+
+```{terminal}
+   :input: pebble services
+Service          Startup  Current   Since
+backend-server   enabled  active    today at 10:20 CST
+database         enabled  inactive  -
+frontend-server  enabled  active    today at 10:20 CST
+```
+
+As we can see, although `database` is in `inactive` state, the backend/frontend are still both started, which isn't exactly useful since all they are doing is consuming resources without being able to provide any service.
+
+Wouldn't it be nice to make these services depend on each other, and when a dependency fails to start, put other services that rely on it on hold, instead of starting them anyway?
+
+Worry no more, service dependency comes to the rescue.
+
+## Service dependencies
+
+Pebble can take service dependencies into account when managing services: the magical `required` (see more detail in [Layer specification](../reference/layer-specification.md)) feature.
```

Comment:
Again, I think "magical" is a bit much here. I get that you're trying to have a fun tone, but I think it strikes the wrong chord for this kind of documentation. It would be fine in a personal blog post.

Also note that it's `requires`, not `required`.

```suggestion
Pebble can take service dependencies into account when managing services: this is done with the `required` list in the [service definition](../reference/layer-specification.md).
```